var/spack/repos/builtin/packages/intel-mpi/package.py

```diff
@@ -62,7 +74,13 @@ def setup_dependent_package(self, module, dep_spec):
         else:
             raise RuntimeError('No suitable bindir found')
 
-        self.spec.mpicc = join_path(bindir, 'mpicc')
```

Comment:
I had a look at:
```console
$ mpiicpc -show
icpc -I/ssoft/spack/external/intel/2017/compilers_and_libraries_2017.2.174/linux/mpi/intel64/include -L/ssoft/spack/external/intel/2017/compilers_and_libraries_2017.2.174/linux/mpi/intel64/lib/release_mt -L/ssoft/spack/external/intel/2017/compilers_and_libraries_2017.2.174/linux/mpi/intel64/lib -Xlinker --enable-new-dtags -Xlinker -rpath -Xlinker /ssoft/spack/external/intel/2017/compilers_and_libraries_2017.2.174/linux/mpi/intel64/lib/release_mt -Xlinker -rpath -Xlinker /ssoft/spack/external/intel/2017/compilers_and_libraries_2017.2.174/linux/mpi/intel64/lib -Xlinker -rpath -Xlinker /opt/intel/mpi-rt/2017.0.0/intel64/lib/release_mt -Xlinker -rpath -Xlinker /opt/intel/mpi-rt/2017.0.0/intel64/lib -lmpicxx -lmpifort -lmpi -lmpigi -ldl -lrt -lpthread
```
when writing this, and Spack compiler wrapper should be called. 

Other than that I agree it may be redundant, but I also think it may be defensive, assuming that variables name like `I_MPI_CC` are more likely to change than wrapper names like `mpiicc`. If people disagree I am fine with rolling this particular modification back.